### PR TITLE
Dosable addr

### LINKED ISSUE #79: deployment addresses: current scheme is DoS-able
These are created client-side and are guessable.  Someone can generate a deployment address for another account for all the sequence numbers.

We need to either verify the account address or think of a new scheme.

### PR DESCRIPTION
**Pull request by aastein**
fixes #79 
refs #164 

- added a nonce field to TxCreateDeployment and TxCreateProvider
- check txs with
  - tenant/owner is a singer
  - nonce is valid
- generate the address on the node in deliverTx
- increment account nonce only after successful deliverTx

### CODE REVIEW

File: types/types.proto [hunk 1]
```diff
@@ -94,6 +96,11 @@ message ProviderAttribute {
   string value = 2;
 }
 
+message GroupSpec {
+  repeated ProviderAttribute requirements = 1 [(gogoproto.nullable) = false];
+  repeated ResourceGroup     resources    = 2 [(gogoproto.nullable) = false];
+}
+
```
Comment (aastein) [inline types/types.proto]: any name suggestions for this?

File: cmd/akash/deployment.go [hunk 2]
```diff
@@ -49,15 +49,27 @@ func createDeploymentCommand() *cobra.Command {
...
-func parseDeployment(file string, tenant []byte, nonce uint64) (*types.Deployment, *types.DeploymentGroups, error) {
+func parseDeployment(file string, nonce uint64) ([]*types.GroupSpec, int64, error) {
 	// todo: read and parse deployment yaml file
 
+	specs := []*types.GroupSpec{}
+
 	/* begin stub data */
-	deployment := testutil.Deployment(tenant, nonce)
-	groups := testutil.DeploymentGroups(deployment.Address, nonce)
+	groups := testutil.DeploymentGroups(*new(base.Bytes), nonce)
```
Comment (boz) [inline cmd/akash/deployment.go]: `testutil.DeploymentAddress(t)` if you want.

File: cmd/akash/deployment.go [hunk 3]
```diff
@@ -98,23 +112,25 @@ func createDeployment(ctx context.Context, cmd *cobra.Command, args []string) er
...
+					fmt.Printf("Group %v/%v Fulfillment: %X\n", f.Group, len(groups),
 						state.FulfillmentID(f.Deployment, f.Group, f.Order, f.Provider))
 				}
 			}).
 			OnTxCreateLease(func(tx *types.TxCreateLease) {
-				if bytes.Equal(tx.Lease.Deployment, deployment.Address) {
+				if bytes.Equal(tx.Lease.Deployment, address) {
 					l := tx.Lease
-					fmt.Printf("Group %v/%v Lease: %X\n", l.Group, len(groups.GetItems()),
+					fmt.Printf("Group %v/%v Lease: %X\n", l.Group, len(groups),
```
Comment (boz) [inline cmd/akash/deployment.go]: hmm not sure how how did `%X` got back in there.  use `"%v",X(state.FulfillmentID(...)))` for these.

File: cmd/akash/provider.go [hunk 4]
```diff
@@ -115,19 +117,19 @@ func doCreateProviderCommand(ctx context.Context, cmd *cobra.Command, args []str
...
 	return nil
 }
 
-func parseProvider(file string, tenant []byte, nonce uint64) (*types.Provider, error) {
+func parseProvider(file string, nonce uint64) ([]types.ProviderAttribute, error) {
 	// todo: read and parse deployment yaml file
 
 	/* begin stub data */
-	provider := testutil.Provider(tenant, nonce)
+	provider := testutil.Provider(*new(base.Bytes), nonce)
```
Comment (boz) [inline cmd/akash/provider.go]: `testutil.Address(t)`

File: types/types.proto (hunk 1 above)
Comment (boz) [reply types/types.proto]: `GroupSpec` seems great.

**Pull request merged**